mongodbatlas/config.go

```diff
@@ -3,18 +3,26 @@ package mongodbatlas
 import (
 	"context"
 	"errors"
+	"fmt"
+	"net/http"
+	"runtime"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
 	"github.com/mongodb-forks/digest"
 	"github.com/mongodb/terraform-provider-mongodbatlas/version"
 	"github.com/spf13/cast"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
 	matlasClient "go.mongodb.org/atlas/mongodbatlas"
 	realmAuth "go.mongodb.org/realm/auth"
 	"go.mongodb.org/realm/realm"
 )
 
-// Config struct ...
+const ToolName = "terraform-provider-mongodbatlas"
+
+var userAgent = fmt.Sprintf("%s/%s (%s;%s)", ToolName, version.ProviderVersion, runtime.GOOS, runtime.GOARCH)
```

Comment:
Very good change to extend the format of the agent.  Before we merge we need change our user agent's reference to cover it and telemetry team to update their search queries

Reply:
Actually, I will revert this change and create a JIRA ticket to do this so that PMs are all aware of this change and we can notify who is using the queries.